example/test/function_component_test.dart

```diff
@@ -183,6 +183,134 @@ UseRefTestComponent(Map props) {
+  final inputRef = useRef();
+
+  useImperativeHandle(
+    ref,
+    () {
+      print('FancyInput: useImperativeHandle re-assigns ref.current');
+      return {'focus': () => inputRef.current.focus()};
+    },
+
+    /// Because the return value of createHandle never changes, it is not necessary for ref.current
+    /// to be re-set on each render so this dependency list is empty.
+    [],
+  );
+
+  return react.input({
+    'ref': inputRef,
+    'value': props['value'],
+    'onChange': (e) => props['update'](e.target.value),
+    'placeholder': props['placeholder'],
+    'style': {'borderColor': props['hasError'] ? 'crimson' : '#999'},
+  });
+});
+
+final useImperativeHandleTestFunctionComponent =
+    react.registerFunctionComponent(UseImperativeHandleTestComponent, displayName: 'useImperativeHandleTest');
+
+UseImperativeHandleTestComponent(Map props) {
+  final city = useState('');
+  final state = useState('');
+  final error = useState('');
+  final message = useState('');
+
+  Ref cityEl = useRef();
+  Ref stateEl = useRef();
+
+  validate(_) {
+    if (!RegExp(r'^[a-zA-Z]+$').hasMatch(city.value)) {
+      message.set('Invalid form!');
+      error.set('city');
+      cityEl.current['focus']();
+      return;
+    }
+
+    if (!RegExp(r'^[a-zA-Z]+$').hasMatch(state.value)) {
+      message.set('Invalid form!');
+      error.set('state');
+      stateEl.current['focus']();
+      return;
+    }
+
+    error.set('');
+    message.set('Valid form!');
+  }
+
+  return react.Fragment({}, [
+    FancyInput({
+      'key': 'fancyInput1',
+      'hasError': error.value == 'city',
+      'placeholder': 'City',
+      'value': city.value,
+      'update': city.set,
+      'ref': cityEl,
+    }, []),
+    FancyInput({
+      'key': 'fancyInput2',
+      'hasError': error.value == 'state',
+      'placeholder': 'State',
+      'value': state.value,
+      'update': state.set,
+      'ref': stateEl,
+    }, []),
+    react.button({'key': 'button1', 'onClick': validate}, ['Validate Form']),
+    react.p({'key': 'p1'}, [message.value]),
+  ]);
+}
+
+final FancyCounter = react.forwardRef((props, ref) {
+  final count = useState(0);
+
+  useImperativeHandle(
+    ref,
+    () {
+      print('FancyCounter: useImperativeHandle re-assigns ref.current');
+      return ({
+        'increment': () => count.setWithUpdater((prev) => prev + props['diff']),
+        'decrement': () => count.setWithUpdater((prev) => prev - props['diff']),
+      });
+    },
+
+    /// This dependency prevents unnecessary calls of createHandle, by only re-assigning
+    /// ref.current when `props['diff']` changes.
+    [props['diff']],
+  );
+
+  return react.div({}, count.value);
+});
+
+final useImperativeHandleTestFunctionComponent2 =
+    react.registerFunctionComponent(UseImperativeHandleTestComponent2, displayName: 'useImperativeHandleTest2');
+
+UseImperativeHandleTestComponent2(Map props) {
+  final diff = useState(1);
+
+  Ref fancyCounterRef = useRef();
```

Comment:
Same as above, this could be typed:
```suggestion
  final fancyCounterRef = useRef<Map>();
```